diagnostic_tool/diagnostic_script.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import blosc
+import dill
+import numpy as np
+from time import sleep, time
+from functools import partial
+from interbotix_xs_modules.xs_robot.arm import InterbotixManipulatorXS
+
+
+if sys.argv[1] == "deploy":
+    DEPLOY_MODE = True
+else:
+    DEPLOY_MODE = False
+
+
+class WX250sManipulator:
+    def __init__(self):
+        self.bot = InterbotixManipulatorXS("wx250s", "arm", "gripper")
+        self.arm = self.bot.arm
+        self.joint_names = self.arm.group_info.joint_names
+        self.num_joints = len(self.joint_names)
+
+    def get_gripper_tf(self):
+        return self.arm.T_sb
+
+    def get_joint_positions(self):
+        return self.arm.core.joint_states.position[:6]
+
+    # def set_joint_positions(self, joint_positions, moving_time):
+    #     pass
+
+    def create_right_angle(self, moving_time=3.0):
+        self.arm.set_joint_positions([0]*self.num_joints, moving_time=moving_time)
```

Comment:
My understanding is that by calling this function, the arm will move to a pose where all joint positions = 0, within exactly 3.0 seconds (or whatever `moving_time` is). Is that right?

This also means that this function is blocking during the movement, or not?

Reply:
This is a dev-file and is no longer exists.